### PR TITLE
fix: only check `instanceof Component` once

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

Previously a check was done for each component type. This simplifies the logic to one check for a class-based component.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

File: packages/builders/src/components/Components.ts [hunk 1]
```diff
@@ -17,13 +17,17 @@ export function createComponent<T extends keyof MappedComponentTypes>(
 ): MappedComponentTypes[T];
 export function createComponent<C extends MessageComponent>(data: C): C;
 export function createComponent(data: APIMessageComponent | MessageComponent): Component {
+	if (data instanceof Component) {
+		return data;
+	}
```
Comment (ImRodry) [inline packages/builders/src/components/Components.ts:22]: Surely you could make this into 1 line right?

Comment (kyranet) [inline packages/builders/src/components/Components.ts:22]:
```suggestion
	if (data instanceof Component) return data;
```

**Review by kyranet — CHANGES_REQUESTED** [review]

**Review by SpaceEEC — APPROVED** [review]

File: packages/builders/src/components/Components.ts (hunk 1 above)
Comment (vladfrangu) [reply packages/builders/src/components/Components.ts:22]: I don't agree with this change 😂

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

File: packages/builders/src/components/Components.ts (hunk 1 above)
Comment (ImRodry) [reply packages/builders/src/components/Components.ts:22]: 🤔

**Review by iCrawl — APPROVED** [review]

**Pull request merged**